src/hough.rs

```diff
@@ -133,7 +134,8 @@ fn intersection_points(
     image_width: u32,
     image_height: u32,
 ) -> Option<((f32, f32), (f32, f32))> {
-    let r = line.r;
+    // FIXME: handle this properly
+    let r = if line.r == 0.0 { line.r + 0.0001 } else { line.r };
```

Comment:
This apparently wants to avoid the cases of division complications and instabilities?

Turning the division into multiplications of the other comparison side, correctly keeping in mind the comparison operator change for `sin(theta) < 0`, might solve this issue. This would effectively simulate the necessary computations in homogeneous coordinates and be more stable.

So for example:

```
let left_y = r * sin.signum();
…
if left_y >= 0 && left_y < h*sin.abs() {
}
```

Reply:
For `(r - w * cos)` itself, `cos.mul_add(-w, r)` could be faster and more accurate as well.

Especially important where `sin` is close to 0, but `w*cos` is also close to `r`, i.e. the line is nearly vertical and near the outermost side of the image.

Reply:
I still need to address this comment. I'm going to rewrite this function and add some tests.

Reply:
It's not right; it was just the quickest hack to test that I'd fixed the bug reported in #280. I've just pushed a commit that removes this line.